Backoffice/EventSubscriber/UpdateNodeCurrentlyPublishedFlagSubscriber.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\EventSubscriber;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use OpenOrchestra\Backoffice\Manager\NodeManager;
+use OpenOrchestra\ModelInterface\Event\NodeEvent;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+use OpenOrchestra\ModelInterface\NodeEvents;
+use OpenOrchestra\ModelInterface\Repository\NodeRepositoryInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class UpdateNodeCurrentlyPublishedFlagSubscriber
+ */
+class UpdateNodeCurrentlyPublishedFlagSubscriber implements EventSubscriberInterface
+{
+    protected $nodeRepository;
+    protected $objectManager;
+
+    /**
+     * @param NodeRepositoryInterface $nodeRepository
+     * @param ObjectManager           $objectManager
+     */
+    public function __construct(NodeRepositoryInterface $nodeRepository, ObjectManager $objectManager)
+    {
+        $this->nodeRepository = $nodeRepository;
+        $this->objectManager = $objectManager;
+    }
+
+    /**
+     * @param NodeEvent $event
+     */
+    public function updateFlag(NodeEvent $event)
+    {
+        $node = $event->getNode();
+
+        if ($node->getStatus()->isPublished()) {
```

Comment:
You could avoid a lot of unnecessaries process if you add a condition here: If the node status is published, but the node is already  currentlyPublished, it's not optimal to do the stuff here, it has always been done previously.

Unless you want to be compatible with older versions of Orchestra, but for this I would privilegiate a doctrine migration script.